### PR TITLE
Base exception, fixes #207

### LINKED ISSUE #207: Add base exception interface

### PR DESCRIPTION
**Pull request by frederikbosch**

### CODE REVIEW

File: src/Exception/Exception.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+namespace Money\Exception;
+
+/**
+ * Base Exception.
+ *
+ * @author Frederik Bosch <f.bosch@genkgo.nl>
+ */
+
+abstract class Exception extends \Exception
```
Comment (sagikazarmark) [inline src/Exception/Exception.php]: I would rather have this as an interface.

Comment (sagikazarmark) [inline src/Exception/Exception.php]: That way we can use SPL exceptions as well and extend whichever we want. For example UnresolvableCurrencyPairException should be invalid argument.

**Comment by frederikbosch** [comment]
Why as interface? You mean like a `marker` interface? What is the difference with an abstract class in thise case?

**Comment by sagikazarmark** [comment]
That way we can use SPL exceptions as well and extend whichever we want. For example UnresolvableCurrencyPairException should be invalid argument:

`UnresolvableCurrencyPairException extends \InvalidArgumentException implements Exception`

**Comment by sagikazarmark** [comment]
:+1:

**Comment by frederikbosch** [comment]
@sagikazarmark Good idea. You do not see that many marker interfaces in PHP.

**Comment by sagikazarmark** [comment]
https://github.com/php-http/httplug/blob/master/src/Exception.php
https://github.com/lstrojny/fxmlrpc/blob/master/src/fXmlRpc/Exception/ExceptionInterface.php
https://github.com/FriendsOfSymfony/FOSHttpCache/blob/master/src/Exception/HttpCacheExceptionInterface.php

There are a few :wink:

**Comment by sagikazarmark** [comment]
Can you squash commits?

**Comment by frederikbosch** [comment]
Merge it in?

**Pull request merged**